### PR TITLE
storage: avoid writing junk data in test

### LINKED ISSUE #2775: log.Fatal in TestStoreRangeSplitAtTablePrefix in multi-cpu
Discovered in #1839

https://circleci.com/gh/cockroachdb/cockroach/8282

https://circle-artifacts.com/gh/cockroachdb/cockroach/8282/artifacts/0/tmp/circle-artifacts.HFeUuK8/test.log

```
F1005 20:41:37.667210 985 storage/replica_command.go:1262  failed to update Store after split: storage/replica_raftstorage.go:378: failed to lookup zone config for Range range=3 ["\xff\f\x03\xe8"-"\xff\xff"): sql/config.go:68: descriptor for object ID 1000 is not a table or database
```

### PR DESCRIPTION
**Pull request by tamird**
Fixes #2775.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3007)

### CODE REVIEW

File: storage/client_split_test.go [hunk 1]
```diff
@@ -100,8 +100,13 @@ func TestStoreRangeSplitAtTablePrefix(t *testing.T) {
 	// Update SystemConfig to trigger gossip.
 	if err := store.DB().Txn(func(txn *client.Txn) error {
 		txn.SetSystemDBTrigger()
+		// We don't care about the values, just the keys.
 		k := sql.MakeDescMetadataKey(sql.ID(keys.MaxReservedDescID + 1))
-		return txn.Put(k, 10)
+		v, err := txn.Get(k)
```
Comment (bdarnell) [inline storage/client_split_test.go]: Will this `Get` ever find anything, or is it always writing an empty string to the given key?

**Comment by bdarnell** [comment]
LGTM, although the point of this test is unclear to me. Doesn't it need to wait for gossip to occur after the transaction commits?

**Comment by tschottdorf** [comment]
Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 3 of 3 files at r3.
Review status: 4 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

<sup>**[storage/client_split_test.go, line 105 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3007#-K2JKtgqbsrdZxG62EHe-r4-105)** ([raw file](https://github.com/cockroachdb/cockroach/blob/613da1cdfe10da808c85bb2c62063125f9b8d1e4/storage/client_split_test.go#L105)):</sup>
@tamird is still working on it right now.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3007)

**Comment by tschottdorf** [comment]
LGTM

**Comment by tamird** [comment]
@bdarnell would you mind taking another look? the test now waits for gossip properly.

**Comment by bdarnell** [comment]
LGTM

**Pull request merged**